addons/mail/static/src/messaging/component/message_list/message_list.js

```diff
@@ -513,6 +518,10 @@ Object.assign(MessageList, {
         haveMessagesAuthorRedirect: Boolean,
         haveMessagesMarkAsReadIcon: Boolean,
         haveMessagesReplyIcon: Boolean,
+        hasScrollAdjust: {
+            type: Boolean,
+            optional: true,
+        },
```

Comment:
If this component is the "final receiver" of the prop, it shouldn't be optional but have a "default false" instead to avoid having 3 values (true, false, undefined). "optional" is desirable only for props that are propagated where undefined makes sense (it means use default value of children).